### ISSUE TITLE
CMake Policy CMP0148

### ISSUE THREAD

**Issue by Ryanf55** [ros2/python_cmake_module#14]
OS: Ubuntu 23.04
CMake version: 3.27.0-rc3
Steps to reproduce: Compile ROS 2 from source on ubuntu 23.04. 
Desired action: Set the cmake policy 
```
CMake Warning (dev) at /home/ryan/ros2_rolling/install/python_cmake_module/share/python_cmake_module/cmake/Modules/FindPythonExtra.cmake:52 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

Call Stack (most recent call first):
  CMakeLists.txt:66 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
```

```
$ cmake --help-policy CMP0148
CMP0148
-------

.. versionadded:: 3.27

The ``FindPythonInterp`` and ``FindPythonLibs`` modules are removed.

These modules have been deprecated since CMake 3.12.
CMake 3.27 and above prefer to not provide the modules.
This policy provides compatibility for projects that have not been
ported away from them.

Projects using the ``FindPythonInterp`` and/or ``FindPythonLibs``
modules should be updated to use one of their replacements:

* ``FindPython3``
* ``FindPython2``
* ``FindPython``

The ``OLD`` behavior of this policy is for ``find_package(PythonInterp)``
and ``find_package(PythonLibs)`` to load the deprecated modules.  The ``NEW``
behavior is for uses of the modules to fail as if they do not exist.

This policy was introduced in CMake version 3.27.  CMake version
3.27.0-rc3 warns when the policy is not set and uses ``OLD`` behavior.
Use the ``cmake_policy()`` command to set it to ``OLD`` or ``NEW``
explicitly.

.. note::
  The ``OLD`` behavior of a policy is
  ``deprecated by definition``
  and may be removed in a future version of CMake.

```

Since these modules are deprecated in ALL versions of current ROS 2 distros, seems like a good reason to fix this warning.

**Comment by mjcarroll** [comment]
Thanks for reporting, appreciate you bravely testing with 23.04 :)

**Comment by clalancette** [comment]
I just merged in #7, which fixes this problem.
